### PR TITLE
Add codspeed benchmarks

### PR DESCRIPTION
**Pull request by bdraco**
## What do these changes do?

Adds codspeed benchmarks

This will get the ball rolling so we know when we have solved #9520

## Are there changes in behavior for the user?

no

## Is it a substantial burden for the maintainers to support this?
no

### CODE REVIEW

File: tests/test_client_request_benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+from pytest_codspeed import BenchmarkFixture  # type: ignore[import-untyped]
+from yarl import URL
+
+from aiohttp.client_reqrep import ClientRequest
+
+
+def test_update_cookies(
+    loop: asyncio.AbstractEventLoop, benchmark: BenchmarkFixture
+) -> None:
```
Comment (bdraco) [inline tests/test_client_request_benchmarks.py]: > This benchmark contains 10 system calls, totalling 93.9 µs of execution time.

That was unexpected

File: tests/test_benchmarks_client_request.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+"""codspeed benchmarks for client requests."""
+
+import asyncio
+from http.cookies import Morsel
+
+from pytest_codspeed import BenchmarkFixture  # type: ignore[import-untyped]
```
Comment (Dreamsorcerer) [inline tests/test_benchmarks_client_request.py:6]: https://github.com/CodSpeedHQ/pytest-codspeed/pull/53

File: .mypy.ini [hunk 3]
```diff
@@ -34,3 +34,13 @@ ignore_missing_imports = True
...
+
+[mypy-test_benchmarks_client_request]
+disable_error_code =
+  no-any-unimported,
+  misc
+
+[mypy-test_benchmarks_cookiejar]
+disable_error_code =
+  no-any-unimported,
+  misc
```
Comment (Dreamsorcerer) [inline .mypy.ini:50]: It's not overly clear why we're disabling these here. Is it a case of being too noisy with pytest-codspeed untyped? In which case we can remove this once they've exposed type annotations.

Comment (bdraco) [reply .mypy.ini:50]: > Is it a case of being too noisy with pytest-codspeed untyped?

That’s exactly the reason.

File: tests/test_benchmarks_client_request.py (hunk 2 above)
Comment (bdraco) [reply tests/test_benchmarks_client_request.py:6]: Thanks!

File: .mypy.ini (hunk 3 above)
Comment (bdraco) [reply .mypy.ini:50]: comments at in 19e6bd302fa43f617456f124c3a067e57b3a7d15

**Pull request merged**